lib/src/rules/unawaited_futures.dart

```diff
@@ -67,13 +68,7 @@ class _Visitor extends SimpleAstVisitor<void> {
 
   @override
   void visitCascadeExpression(CascadeExpression node) {
-    for (var expr in node.cascadeSections) {
-      if ((expr.staticType?.isDartAsyncFuture ?? false) &&
-          _isEnclosedInAsyncFunctionBody(expr) &&
-          expr is! AssignmentExpression) {
-        rule.reportLint(expr);
-      }
-    }
+    node.cascadeSections.forEach(_visit);
```

Comment:
Although it's textually cleaner in this case, `forEach` is really slow. A for-in loop is a little better, but a normal for loop is by far the fastest way to iterate:
```
var sections = node.cascadeSections;
for (int i = 0; i < sections.length; i++) {
  _visit(sections[i]);
}
```
I don't know how important the performance difference is in this case, but I'm starting to think that we need to be more aware of performance in lint code.

Reply:
Thanks, done.